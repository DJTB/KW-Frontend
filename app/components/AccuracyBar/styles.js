import styled from 'styled-components';
import { transparentize, darken } from 'polished';

import * as COLORS from 'shared/styles/colors';
import { gutter } from 'shared/styles/layout';
import { beta } from 'shared/styles/typography';

export const Wrapper = styled.div`
  ${gutter()};
  flex: 1 1 auto;
`;

export const Background = styled.div`
  display: flex;
  position: relative;
  padding: .8rem;
  flex: 1 1 auto;
  background-color: ${({ bgColor }) => transparentize(0.75, COLORS[bgColor])};
`;

export const Text = styled.h1`
  ${beta}
  margin: 0;
  line-height: 1;
  align-self: center;
  color: ${COLORS.whiteLight};
  text-shadow: 1px 1px 0.1em ${({ textShadowColor }) => transparentize(0.5, darken(0.4, COLORS[textShadowColor]))};
  z-index: 2;
`;

export const Bar = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  z-index: 1;
`;
