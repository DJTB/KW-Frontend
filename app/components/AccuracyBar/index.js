import React from "react";
import PropTypes from "prop-types";
import { Motion, spring } from "react-motion";

import * as COLORS from "shared/styles/colors";

import { Wrapper, Background, Text, Bar } from "./styles";

AccuracyBar.propTypes = {
  percent: PropTypes.number,
  color: PropTypes.oneOf(Object.keys(COLORS)),
};

AccuracyBar.defaultProps = {
  percent: 0,
  color: "purple",
};

function AccuracyBar({ percent, color }) {
  return (
    <Wrapper>
      <Background bgColor={color}>
        <Text percent={percent} textShadowColor={color}>
          {percent}% Accuracy
        </Text>
        <Motion defaultStyle={{ percentX: 0 }} style={{ percentX: spring(percent) }}>
          {({ percentX }) => (
            <Bar
              style={{ backgroundColor: COLORS[color] || color, width: `${percentX}%` }}
            />
          )}
        </Motion>
      </Background>
    </Wrapper>
  );
}

export default AccuracyBar;
