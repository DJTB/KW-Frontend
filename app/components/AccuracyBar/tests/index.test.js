import React from 'react';
import { shallow } from 'enzyme';
import AccuracyBar from '../index';

describe('<AccuracyBar />', () => {
  it('should match baseline snapshot', () => {
    expect(shallow(<AccuracyBar />)).toMatchSnapshot();
  });
  it('should adopt percent and color props', () => {
    expect(shallow(<AccuracyBar percent={24} color="green" />)).toMatchSnapshot();
  });
});
