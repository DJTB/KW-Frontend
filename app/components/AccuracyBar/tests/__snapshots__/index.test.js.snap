// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccuracyBar /> should adopt percent and color props 1`] = `
<styled.div>
  <styled.div
    bgColor="green"
  >
    <styled.h1
      percent={24}
      textShadowColor="green"
    >
      24
      % Accuracy
    </styled.h1>
    <styled.div
      bgColor="green"
      percent={24}
    />
  </styled.div>
</styled.div>
`;

exports[`<AccuracyBar /> should match baseline snapshot 1`] = `
<styled.div>
  <styled.div
    bgColor="purple"
  >
    <styled.h1
      percent={0}
      textShadowColor="purple"
    >
      0
      % Accuracy
    </styled.h1>
    <styled.div
      bgColor="purple"
      percent={0}
    />
  </styled.div>
</styled.div>
`;
